src/main/java/org/broadinstitute/hellbender/utils/hdf5/HDF5PoNCreator.java

```diff
@@ -0,0 +1,712 @@
+        } else {
+            final double[] singularValues = logNormalsSVD.getSingularValues();
+            final double mean = MathUtils.mean(singularValues, 0, singularValues.length);
+            final double eigenValueCutoff = mean * JOLLIFES_RULE_MEAN_FACTOR; // Jollife's less strict version of Kaiser' Rule.
+            numberOfEigenSamples = (int) DoubleStream.of(singularValues).filter(v -> v > eigenValueCutoff).count();
+            logger.info(String.format("Jollife's rule produced %d eigen samples out of %d possibles for the reduced PoN", numberOfEigenSamples, numberOfCountColumns));
+        }
+        return numberOfEigenSamples;
+    }
+
+    /**
+     * Calculates the median of column medians and subtract it from all counts.
+     * @param readCounts the input counts to center.
+     * @return the median of medians that has been subtracted from all counts.
+     */
+    @VisibleForTesting
+    static double subtractBGSCenter(final ReadCountCollection readCounts, final Logger logger) {
+        final RealMatrix counts = readCounts.counts();
+        final int columnCount = counts.getColumnDimension();
+        final double[] columnMedians = new double[columnCount];
+        final double[][] data = counts.getData();
+        final Median medianCalculator = new Median();
+        final double[] columnData = new double[data.length];
+        for (int i = 0; i < columnCount; i++) {
+            for (int j = 0; j < data.length; j++) {
+                columnData[j] = data[j][i];
+            }
+            columnMedians[i] = medianCalculator.evaluate(columnData);
+        }
+        final double medianOfMedians = medianCalculator.evaluate(columnMedians);
+        counts.walkInOptimizedOrder(new DefaultRealMatrixChangingVisitor() {
+            @Override
+            public double visit(final int row, final int column, final double value) {
+                return value - medianOfMedians;
+            }
+        });
+        logger.info(String.format("Counts centered around the BGS center %.2f", medianOfMedians));
+        return medianOfMedians;
+    }
+
+    /**
+     * Final pre-panel normalization that consists on dividing all counts by the median of
+     * its column and log it with base 2.
+     * <p>
+     *     The normalization occurs in-place.
+     * </p>
+     *
+     * @param readCounts the input counts to normalize.
+     */
+    @VisibleForTesting
+    static void normalizeAndLogReadCounts(final ReadCountCollection readCounts, final Logger logger) {
+        final RealMatrix counts = readCounts.counts();
+        final int columnCount = counts.getColumnDimension();
+        final Median medianCalculator = new Median();
+        for (int i = 0; i < columnCount; i++) {
+            final double[] colCounts = counts.getColumn(i);
+            final double median = medianCalculator.evaluate(colCounts);
+            final double invMedian = 1.0 / median;
+            for (int j = 0; j < colCounts.length; j++) {
```

Comment:
This could be handled with a visitor:
```
final double[] medians = IntStream.range(0, counts.getColumnDimension()).map(col -> medianCalculator.evaluate(counts.getColumn(i));
counts.walkInOptimizedOrder(new DefaultRealMatrixChangingVisitor() {
            @Override
            public double visit(final int row, final int column, final double value) { return value / medians[column]; }
        });
```